packages/x-ui-collections-bundle/src/graphql/Collection.ts

```diff
@@ -749,4 +749,31 @@ export abstract class Collection<T = null> {
   protected isUpdateModifier(element: any): element is UpdateFilter<T> {
     return element["$set"];
   }
+
+  /**
+   * Integration with native `useQuery` from Apollo, offering a hybrid approach so apollo re-uses your cache.
+   */
+  public optimistic(_id: ObjectId | string, document: T) {
+    const stringId = _id instanceof ObjectId ? _id.toString() : _id;
+    const operationName = ""; // ! how do we determine this here ?
+
+    // We can't be optimistic if we do no know the operationName
+    if (!operationName) return undefined;
+
+    const collectionName = this.getName?.();
+
+    // We can't be optimistic if we do no know the __typename
+    if (!collectionName) return undefined;
+
+    return {
+      [operationName]: {
+        _id: stringId,
+        __typename:
+          collectionName[collectionName.length - 1] === "s"
+            ? collectionName.slice(0, -1)
+            : collectionName, // we could use https://www.npmjs.com/package/pluralize
+        ...document, // should we deepclone here just in case ?
```

Comment:
I think you need to transform the document into a `graph-like` model of it.

Reply:
related to #87

Reply:
I'm not sure of hat you mean.
What we need here is to provide the exact same values as in the input : https://www.apollographql.com/docs/react/performance/optimistic-ui/